Communication/Protocols/UCI/Options/SpinOption.cs

```diff
@@ -0,0 +1,40 @@
+namespace Communication.Protocols.UCI
+{
+    class SpinOption : BaseOption
+    {
+        private int minAllowed;
+        private int maxAllowed;
+        private int value;
+
+        public SpinOption(string name, int minAllowed, int maxAllowed, int value) : base(name, "spin")
+        {
+            if (minAllowed > maxAllowed)
+            {
+                throw new UCIProtocolException($"Minimum allowed value '{minAllowed}' cannot be bigger then the maximum value '{maxAllowed}'.");
+            }
+
+            this.minAllowed = minAllowed;
+            this.maxAllowed = maxAllowed;
+            SetValue(value);
+        }
+
+        public void SetValue(int value)
+        {
+            EnsureInAllowedRange(value);
+            this.value = value;
+        }
+
+        private void EnsureInAllowedRange(int value)
+        {
+            if (value < minAllowed || value > maxAllowed)
+            {
+                throw new UCIProtocolException("$Option value '{value}' must be in range [{this.minAllowed}, {this.maxAllowed}].");
```

Comment:
Is '$' sign placed correctly? Probably outside of  the string

Reply:
Fixed.